rmw_opendds_cpp/src/OpenDDSNode.cpp

```diff
@@ -140,6 +141,9 @@ OpenDDSNode::OpenDDSNode(rmw_context_t & context)
     if (!sub_listener_) {
       throw std::runtime_error("CustomSubscriberListener failed");
     }
+    OpenDDS::DCPS::Discovery_rch disco = TheServiceParticipant->get_discovery(context.options.domain_id);
+    OpenDDS::RTPS::RtpsDiscovery_rch rtps_disco = OpenDDS::DCPS::dynamic_rchandle_cast<OpenDDS::RTPS::RtpsDiscovery>(disco);
+    rtps_disco->use_xtypes(false);
     dp_ = context.impl->dpf_->create_participant(
```

Comment:
If the call to rtps_disco->use_xtypes(false) is domain_id specific, not participant specific, these few lines should be moved to rmw_context_impl_t constructor right after TheServiceParticipant->set_default_discovery(OpenDDS::DCPS::Discovery::DEFAULT_RTPS). In that way, it's guaranteed to set xtypes false before creating any participant, and also to avoid calling it repeatedly for each participant.

Reply:
https://github.com/oci-labs/rmw_opendds/blob/system_test_cleanup/rmw_opendds_cpp/src/init.cpp

Reply:
That may involve adding domain ID in struct rmw_context_impl_t, since the domain_id is default initialized (RMW_DEFAULT_DOMAIN_ID) in rmw_init_options_t.